atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -100,12 +102,59 @@
 import com.palantir.lock.impl.LegacyTimelockService;
 import com.palantir.lock.impl.LockServiceImpl;
 import com.palantir.lock.v2.TimelockService;
-import com.palantir.logsafe.UnsafeArg;
 import com.palantir.timestamp.TimestampService;
 import com.palantir.timestamp.TimestampStoreInvalidator;
 import com.palantir.util.OptionalResolver;
 
-public final class TransactionManagers {
+@Value.Immutable
+public abstract class TransactionManagers {
+    abstract AtlasDbConfig config();
+
+    @Value.Default
+    Supplier<Optional<AtlasDbRuntimeConfig>> runtimeConfigSupplier() {
+        return Optional::empty;
+    }
+
+    abstract Set<Schema> schemas();
+
+    @Value.Default
+    Consumer<Object> env() {
```

Comment:
nit: should we rename this? env has felt weird outside context of dropwizard

Reply:
I call it `registrar` or `resourceRegistrar` in other places where I've done similar things